pkg/operator/ceph/webhook.go

```diff
@@ -36,29 +36,72 @@ import (
 const (
 	admissionControllerAppName       = "rook-ceph-admission-controller"
 	tlsPort                    int32 = 443
+	webhookEnv                       = "ROOK_DISABLE_ADMISSION_CONTROLLER"
 )
 
 var (
-	namespace = os.Getenv(k8sutil.PodNamespaceEnvVar)
+	namespace              = os.Getenv(k8sutil.PodNamespaceEnvVar)
+	certManagerWebhookName = "cert-manager-webhook"
 )
 
-func isSecretPresent(ctx context.Context, context *clusterd.Context) (bool, error) {
+func createWebhook(ctx context.Context, context *clusterd.Context) (bool, error) {
+	certMgrClient, err := cs.NewForConfig(context.KubeConfig)
+	if err != nil {
+		logger.Errorf("failed to set config for cert-manager. %v", err)
+		return false, nil
+	}
+
+	if os.Getenv(webhookEnv) == "true" {
+		logger.Info("delete Issuer and Certificate since secret is not found")
+		if err = deleteIssuerAndCetificate(ctx, certMgrClient, context); err != nil {
+			logger.Errorf("failed to delete issuer or certificate. %v", err)
+		}
+		return false, nil
+	}
+
+	logger.Infof("Fetching webhook %s to see if cert-manager is installed.", certManagerWebhookName)
+	_, err = context.Clientset.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(ctx, certManagerWebhookName, metav1.GetOptions{})
+	if err != nil {
+		logger.Info("failed to get cert manager")
+		return false, nil
+	}
+
+	issuer, err := fetchorCreateIssuer(ctx, certMgrClient)
+	if err != nil {
+		logger.Errorf("issuer creation failed %v", err)
+		return false, nil
+	}
+
+	err = fetchorCreateCertificate(ctx, certMgrClient, issuer)
+	if err != nil {
+		logger.Errorf("certificate creation failed %v", err)
+		return false, nil
+	}
+
 	logger.Infof("looking for admission webhook secret %q", admissionControllerAppName)
 	s, err := context.Clientset.CoreV1().Secrets(namespace).Get(ctx, admissionControllerAppName, metav1.GetOptions{})
 	if err != nil {
 		// If secret is not found. All good ! Proceed with rook without admission controllers
+		logger.Info("delete Issuer and Certificate since secret is not found")
+		if err = deleteIssuerAndCetificate(ctx, certMgrClient, context); err != nil {
```

Comment:
Seems like we should only delete the issuer and cert if the error is not found. Shall we move this deletion inside the IsNotFound check of line 89?

Reply:
if we failed to get secret webhook will not start and the resources will be there but I agree we should me deleting only when error is not found because maybe in next reconcile it might be present or at least we'll return the error and not just delete the resources.